source/extensions/omni.isaac.lab/omni/isaac/lab/envs/mdp/events.py

```diff
@@ -1004,3 +1004,46 @@ def _randomize_prop_by_op(
             f"Unknown operation: '{operation}' for property randomization. Please use 'add', 'scale', or 'abs'."
         )
     return data
+
+def randomize_body_coms(
+    env: ManagerBasedEnv, 
+    env_ids: torch.Tensor | None, 
+    com_range: dict[str, tuple[float, float]], 
+    asset_cfg: SceneEntityCfg
+):
+    """Randomize the CoM of the bodies by adding a random value sampled from the given range.
+
+    .. note::
+        This function uses CPU tensors to assign the CoM. It is recommended to use this function
+        only during the initialization of the environment.
+    """
+    # extract the used quantities (to enable type-hinting)
+    asset: Articulation = env.scene[asset_cfg.name]
+
+    # resolve environment ids
+    if env_ids is None:
+        env_ids = torch.arange(env.scene.num_envs, device="cpu")
+    else:
+        env_ids = env_ids.cpu()
+
+    # resolve body indices
+    if asset_cfg.body_ids == slice(None):
+        body_ids = torch.arange(asset.num_bodies, dtype=torch.int, device="cpu")
+    else:
+        body_ids = torch.tensor(asset_cfg.body_ids, dtype=torch.int, device="cpu")
+
+    # sample random CoM values
+    range_list = [com_range.get(key, (0.0, 0.0)) for key in ["x", "y", "z"]]
+    ranges = torch.tensor(range_list, device="cpu")
+    rand_samples = math_utils.sample_uniform(ranges[:, 0], ranges[:, 1], (len(env_ids), 3), device="cpu").unsqueeze(1)
```

Comment:
This should be num bodies too?

```suggestion
    rand_samples = math_utils.sample_uniform(ranges[:, 0], ranges[:, 1], (len(env_ids), len(body_ids), 3), device="cpu").unsqueeze(1)
```